packages/qwik/src/core/signal/signal.ts

```diff
@@ -345,15 +345,12 @@ export const triggerEffects = (
             container.$scheduler$(ChoreType.QRL_RESOLVE, null, effect.$computeQrl$);
           }
         }
-        (effect as ComputedSignal<unknown> | WrappedSignal<unknown>).$invalid$ = true;
-        const previousSignal = signal;
-        try {
-          signal = effect;
-          effect.$effects$?.forEach(scheduleEffect);
-        } catch (e: unknown) {
-          logError(e);
-        } finally {
-          signal = previousSignal;
+        if (effect instanceof ComputedSignal || effect instanceof WrappedSignal) {
+          try {
+            retryOnPromise(() => effect.$invalidate$());
```

Comment:
Should force() not also use this API then?

Reply:
`force` don't use computeIfNeeded so probably not needed